### PR TITLE
Allow actions with dot prefix; fixes #780

### LINKED ISSUE #780: Qualified names in actionRef syntax?
Not sure this is a bug but I'm confused by this example in [section 13.2.1.2](https://p4.org/p4-spec/docs/P4-16-v1.1.0-spec.html#sec-table-action-list).
> Each action in the list of actions for a table must have a distinct name—e.g., the following program fragment is illegal:
>
>```action a() {}
>control c() {
>    action a() {}
>    // Illegal table: two actions with the same name
>    table t { actions = { a; .a; } }
>}

p4c rejected the example with a parse error, rather than reporting something about actions with  a shared name. I tried modifying the example to only have `actions = { .a; }`, but that was also rejected by p4c. After taking a look at the grammar it seems like actionRefs are bare names, rather than qualified names. I was misled by the example, I think--maybe it could be removed or the syntax for actionRefs could be expanded?

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -5724,8 +5724,8 @@ actionList
     ;
 
 actionRef
-    : optAnnotations name
-    | optAnnotations name '(' argumentList ')'
+    : optAnnotations prefixedNonTypeName
```
[the remark this answers is not in the document]
Comment (jnfoster) [reply p4-16/spec/P4-16-spec.mdk]: Provided we respect lexical scope, I think top-level or not is no different?

[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk]: No, at this point the grammar is 
```yacc
prefixedNonTypeName
    : nonTypeName
    | dotPrefix nonTypeName
```

[the remark this answers is not in the document]
Comment (vgurevich) [reply p4-16/spec/P4-16-spec.mdk]: So, now it is just `a()` and `.a()` or something else? Do we really need `.a()` -- perhaps it's better to simply respect the scope and be done with it?

[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk]: We will have to discuss this in the LDWG.

[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk]: We will have to discuss this in the LDWG.

**Pull request merged**